static/app/components/events/interfaces/spans/spanGroupBar.tsx

```diff
@@ -142,7 +144,45 @@ function renderMeasurements(
 
 export function SpanGroupBar(props: Props) {
   const spanTitleRef: LegacyRef<HTMLDivElement> | null = useRef(null);
-  const {onWheel} = props;
+
+  const {
+    onWheel,
+    generateContentSpanBarRef,
+    getScrollLeftValue,
+    didAnchoredSpanMount,
+    spanGrouping,
+    toggleSpanGroup,
+  } = props;
+
+  // On mount, it is necessary to set the left styling of the content here due to the span tree being virtualized.
+  // If we rely on the scrollBarManager to set the styling, it happens too late and awkwardly applies an animation.
+
+  const setTransformCallback = useCallback(
+    ref => {
+      generateContentSpanBarRef()(ref);
+      if (ref) {
+        const left = -getScrollLeftValue();
+        ref.style.transform = `translateX(${left}px)`;
+        ref.style.transformOrigin = 'left';
+      }
+    },
+    [generateContentSpanBarRef, getScrollLeftValue]
+  );
+
+  useEffect(() => {
+    if (location.hash && !didAnchoredSpanMount) {
+      const anchoredSpanIndex = spanGrouping.findIndex(
+        span => spanTargetHash(span.span.span_id) === location.hash
+      );
+
+      // TODO: This doesn't always work.
+      // A potential fix is to just scroll to the Autogroup without expanding it if a span within it is anchored.
+      if (anchoredSpanIndex > -1) {
+        toggleSpanGroup();
+        window.scrollTo(0, window.scrollY + anchoredSpanIndex * ROW_HEIGHT);
+      }
+    }
+  }, [didAnchoredSpanMount, spanGrouping, toggleSpanGroup]);
```

Comment:
This effect determines whether a span inside this grouping is anchored. If it is, then it will expand the group and try to scroll to the estimated position of the anchored span within it. Sadly, it doesn't always work :/

I'm going to come back to this later since this works in most cases, and it's difficult to reproduce the conditions when it fails. It should be fine in the majority of cases.